core/libp2p/discovery/peer_discovery.hpp

```diff
@@ -0,0 +1,22 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_PEER_DISCOVERY_HPP
+#define KAGOME_PEER_DISCOVERY_HPP
+
+#include <rxcpp/rx-observable.hpp>
+#include "libp2p/common/peer_info.hpp"
+
+namespace libp2p::discovery {
+  class PeerDiscovery {
+    /**
+     * Return observable to peers, discovered by this module
+     * @return observable to new peers
+     */
+    virtual rxcpp::observable<common::PeerInfo> getNewPeers() const = 0;
```

Comment:
Is that interface is enough? I'm pretty sure that variant with peers caching is more useful. So interface might just return some structure reference with the peer list

Reply:
This method is for returning peers, which are discovered by the system, in real-time, so that, for example, if we want to establish connection with each new one, we can do it. But, this class really misses a method to return all cached peers, so I will add it